src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -586,22 +651,105 @@ def _process_genome_objects(
     loader_helper.create_softlink_between_dirs(new_dir, target_dir)
 
 
-def _upload_assembly_files_in_parallel(
-    asu: AssemblyUtil,
-    ws: Workspace,
-    upload_env_key: str,
-    workspace_id: int,
-    load_id: str,
-    collections_source_dir: str,
-    wait_to_upload_objs: dict[str, str],
-    batch_size: int,
-    source_dir: str,
+def _process_batch_upload(
+        obj_tuples: list[WSObjTuple],
+        conf: Conf,
+        workspace_id: int,
+        load_id: str,
+        service_ver: str,
+        upload_assembly_only: bool = True,
+) -> list[UploadResult]:
+    if upload_assembly_only:
+        asu = AssemblyUtil(conf.callback_url, service_ver=service_ver, token=conf.token)
+        upload_results = _upload_assemblies_to_workspace(asu, workspace_id, load_id, obj_tuples)
+    else:
+        gfu = GenomeFileUtil(conf.callback_url, service_ver=service_ver, token=conf.token)
+        upload_results = _upload_genomes_to_workspace(gfu, workspace_id, load_id, obj_tuples, conf.job_data_dir)
+
+    return upload_results
+
+
+def _process_failed_uploads(
+        ws: Workspace,
+        workspace_id: int,
+        load_id: str,
+        obj_tuples: list[WSObjTuple],
+        upload_assembly_only: bool = True
+) -> list[UploadResult]:
+
+    # figure out uploads that succeeded
+    upload_results = list()
+    name2tuple = {
+        obj_tuple.obj_name: obj_tuple
+        for obj_tuple in obj_tuples
+    }
+    (uploaded_obj_names,
+     uploaded_obj_upas,
+     uploaded_assembly_obj_infos,
```

Comment:
These also seem redundant come to think of it

Reply:
👍

Reply:
:+1: